src/accountsdb/db.zig

```diff
@@ -1235,17 +1253,26 @@ pub const AccountsDB = struct {
                     // hashes aren't always stored correctly in snapshots
                     if (account_hash.eql(Hash.ZEROES)) {
                         const account, var lock_guard = try self.getAccountFromRefWithReadLock(max_slot_ref);
```

Comment:
ticket for fixing/optimizing `getAccountHashAndLamportsFromRef` (not in this diff but near this location) would be good too since its the main bottleneck (including removing the bufferpool from this usage since the data is thrown away immediately after and we just need to read the file directly)

Reply:
#538